gpflow/likelihoods/base.py

```diff
@@ -340,10 +397,10 @@ def _variational_expectations(self, Fmu, Fvar, Y):
         :param Y: observation Tensor, with shape [..., latent_dim]:
         :returns: variational expectations, with shape [...]
         """
-        return tf.reduce_sum(
-            ndiagquad(self._scalar_log_prob, self.num_gauss_hermite_points, Fmu, Fvar, Y=Y),
-            axis=-1,
-        )
+        if self.latent_dim is None:
+            self.latent_dim = int(Fmu.shape[-1])
+            self.observation_dim = self.latent_dim
```

Comment:
Following the DRY-principle it would make sense to add a `_set_latent_and_observation_dimension_eagerly(self, Fmu)` function to this class.

Reply:
Done!